### PR TITLE
MAT-9904: Update endpoints to work with Validator

### PR DESCRIPTION
**Pull request by chubert-sb**
## Terminology Service PR

Jira Ticket: [MAT-9904](https://jira.cms.gov/browse/MAT-9904)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/terminology/config/SecurityConfig.java [hunk 1]
```diff
@@ -16,7 +16,9 @@
 @EnableMethodSecurity
 public class SecurityConfig {
 
-  private static final String[] AUTH_WHITELIST = {"/actuator/**"};
+  private static final String[] AUTH_WHITELIST = {
+    "/actuator/**", "/terminology/ValueSet", "/terminology/CodeSystem"
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/terminology/config/SecurityConfig.java:20]: We could name the new controller with @RequestMapping(path = "/vses")  instead of @RequestMapping(path = "/terminology")

**Review by RohitKandimalla — DISMISSED** [review]

File: src/main/java/gov/cms/madie/terminology/controller/VsacControllerAdvice.java [hunk 2]
```diff
@@ -185,7 +185,7 @@ Map<String, Object> onCodeSystemNotFoundException(
   }
 
   @ExceptionHandler(ValueSetNotFoundException.class)
-  @ResponseStatus(HttpStatus.NOT_FOUND)
+  @ResponseStatus(HttpStatus.NO_CONTENT)
```
Comment (nmorasb) [inline src/main/java/gov/cms/madie/terminology/controller/VsacControllerAdvice.java:188]: does changing this response status have any regression impacts on the UI?

File: src/main/java/gov/cms/madie/terminology/config/SecurityConfig.java (hunk 1 above)
Comment (chubert-sb) [reply src/main/java/gov/cms/madie/terminology/config/SecurityConfig.java:20]: Apparently those mappings are limited. I've talked to Brendan about that before. It's a lot easier on devOps to stick them under a common path like terminology

File: src/main/java/gov/cms/madie/terminology/config/ApiKeyFilter.java [hunk 3]
```diff
@@ -46,16 +47,48 @@ protected void doFilterInternal(
...
+      String[] parts = credentials.split(":", 2);
+      if (parts.length != 2) {
+        sendUnauthorized(response);
+        return;
+      }
+
+      String password = parts[1];
+
+      // Validate against your expected API key (you can use username or password)
+      // For example, if API key is just the password:
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/terminology/config/ApiKeyFilter.java]: "Example" Comment doesn't seem relevant.

File: src/main/java/gov/cms/madie/terminology/config/SecurityConfig.java (hunk 1 above)
Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/terminology/config/SecurityConfig.java:20]: Correct. For reference, It's a money saving choice. We get a limited number of paths for all services at our current pricing tier. Going up would give us many more paths, but also increase in price dramatically.

Hence, we have elected to start all request paths with a service identifier.

File: src/main/java/gov/cms/madie/terminology/controller/VsacControllerAdvice.java (hunk 2 above)
Comment (chubert-sb) [reply src/main/java/gov/cms/madie/terminology/controller/VsacControllerAdvice.java:188]: It shouldn't. This exception was made as part of the VSES work. I checked and didn't see any other usage in the UI. I mostly changed the status to align better with HAPI Fhir validator, so that it would continue the chain after this response.

File: src/main/java/gov/cms/madie/terminology/config/ApiKeyFilter.java (hunk 3 above)
Comment (chubert-sb) [reply src/main/java/gov/cms/madie/terminology/config/ApiKeyFilter.java]: AI put that there. removed

**Review by jkotanchik-SB — APPROVED** [review]
